api/search.go

```diff
@@ -309,3 +332,54 @@ func sanitiseDoubleQuotes(str string) string {
 	b := strconv.Quote(str)
 	return b[1 : len(b)-1]
 }
+
+func processSearchQuery(ctx context.Context, elasticSearchClient DpElasticSearcher, queryBuilder QueryBuilder, sanitisedQuery, typesParam, sort string, topicSlice []string, limit, offset int, responseDataChan chan []byte) {
+	formattedQuery, err := queryBuilder.BuildSearchQuery(ctx, sanitisedQuery, typesParam, sort, topicSlice, limit, offset, true)
+	if err != nil {
+		log.Error(ctx, "creation of search query failed", err, log.Data{"q": sanitisedQuery, "sort": sort, "limit": limit, "offset": offset})
+		responseDataChan <- nil
+	}
+
+	var searches []client.Search
+
+	err = json.Unmarshal(formattedQuery, &searches)
```

Comment:
Collapse line 345 and 346 together:

```suggestion
	if err = json.Unmarshal(formattedQuery, &searches); err != nil {
```

Reply:
Fixed in update 1.